diff --git a/src/node_core_utils.js b/src/node_core_utils.js
index e4e217f9cc26cc5a3df99083ac742268fa1c3ae1..3e4d0757934e600376ab4431768f96fd42a6a1e2 100644
--- a/src/node_core_utils.js
+++ b/src/node_core_utils.js
@@ -49,17 +49,18 @@ function verifyAuth(signStr, streamId, secretKey) {
 function getFFmpegVersion(ffpath) {
   return new Promise((resolve, reject) => {
     let ffmpeg_exec = spawn(ffpath, ['-version']);
-    let version = '';
+    let result = '';
     ffmpeg_exec.on('error', (e) => {
       reject(e);
     });
     ffmpeg_exec.stdout.on('data', (data) => {
       try {
-        version = data.toString().split(/(?:\r\n|\r|\n)/g)[0].split('\ ')[2];
+        result += data;
       } catch (e) {
       }
     });
     ffmpeg_exec.on('close', (code) => {
+      const version = result.toString().split(/(?:\r\n|\r|\n)/g)[0].split('\ ')[2];
       resolve(version);
     });
   });