diff --git a/src/api/controllers/streams.js b/src/api/controllers/streams.js
index 5548feba684ca7dc570f362e0d14de9ede7c703e..28a26682b8bc3fce06d5e59a4997997c78690c58 100644
--- a/src/api/controllers/streams.js
+++ b/src/api/controllers/streams.js
@@ -120,7 +120,8 @@ function getStream(req, res, next) {
     arguments: {}
   };
 
-  let publishStreamPath = `/${req.params.app}/${req.params.stream}`;
+  const cleanStream = encodeURIComponent(req.params.stream);
+  let publishStreamPath = `/${req.params.app}/${cleanStream}`;
 
   let publisherSession = this.sessions.get(
     this.publishers.get(publishStreamPath)